Add mlr3pipelines framework comparison to Appendix

pfistfl opened the pull request.

pat-s commented on bookdown/98-appendix.Rmd:
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
+
+The following example obtained from the **sklearn** documentation showcases a **Pipeline**
+that first Selects a feature and performs PCA on the original data, concatenates the resulting datasets
+and applies a Support Vector Machine.
```
- watch package linking
- watch linebreaking

pat-s commented on bookdown/98-appendix.Rmd:
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
+In summary, we can achieve similar results with a comparable number of lines, while at the
+same time offering greater flexibility with respect to which kinds of pipelines we want to
+optimize over. At the same time, experiments using `mlr3` can now be arbitrarily parallelized
+using `futures`.
```
- watch linebreaking
- watch package linking

pat-s commented on bookdown/98-appendix.Rmd:
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
+methods using a pipe-operator.
+As `recipes` package tightly integrates with the `tidymodels` ecosystem, much of the
+functionality integrated there can be used in `recipes`.
+We compare `recipes` to `mlr3pipelines` using an example from the `recipes` vignette.
```
- watch linebreaking
- watch package linking

pat-s commented on bookdown/98-appendix.Rmd:
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
+information on the customers. In order to do this, we build a model that does the following:
+1. It first imputes missing values using k-nearest neighbours
+2. All factor variables are converted to numerics using dummy encoding
+3. The data is first centered then scaled.
```
- watch linebreaking
- list should have an empty line before

pat-s commented on bookdown/98-appendix.Rmd:
```diff
@@ -89,3 +89,360 @@ as.data.frame(as.data.table(mlr_filters)) %>%
+
+In order to validate the algorithm, data is first split into a train and test set using
+`initial_split`, `training`, `testing`. The recipe trained on the train data (see steps above)
+is then applied to the test data.
```
watch linebreaking

pfistfl commented:
I think I have everything.

- Added mlr3pipelines table. I think this is better than nothing for now
- Repaired the vignette to concur with style-guide
- Added a few lines to the pipeline's section in order to introduce sugar appropriately
